server.js

```diff
@@ -133,3 +147,96 @@ async function pinningNode () {
     console.log('peer joined room', topic, peer)
   }
 }
+
+pinningNode()
+
+
+
+const getProfile = async (rootStoreAddress) => {
+  // const profile = getProfile(req.params.address)
+  const rootStore = await orbitdb.open(rootStoreAddress)
+  const readyPromise = new Promise((resolve, reject) => {
+    rootStore.events.on('ready', resolve)
+  })
+  rootStore.load()
+  await readyPromise
+
+  await Promise.resolve(resolve => {
+    rootStore.events.on('replicated', resolve)
```

Comment:
Why do we wait for this event? replicated is only emitted if you replicate from the network which we won't do here.

Reply:
Ok this I'm also wondering, pretty sure I originally removed it, things didn't work and then added it back in. And then saw no time lost in opening and fetching the profile. So not sure exactly what it is doing

Reply:
But I will take a closer look to pinpoint or add details for a future issue

Reply:
that's odd. let's keep it then

Reply:
ok, ty!